### PR TITLE
Add metrics spec to example deployments to use with Prometheus/Grafana

### PR DESCRIPTION
**Pull request by edgarlanting**
This example files PR adds the needed "internals" used for our Grafana/Prometheus setup. Currently, this is only being added for a simple cluster and a production cluster deployment but this can easily be added to other types of deployments. Adding it to the examples gives customers an option to easier roll it out.

Added functionality:

```
metrics:
    mode: sidecar
    enabled: true
    image: arangodb/arangodb-exporter:0.1.7
    tls: false
  annotations:                                                                                                                                                  
    prometheus.io/scrape: 'true'                                                                                                                               
    prometheus.io/port: '9101'                                                                                                                                 
    prometheus.io/scrape_interval: '5s'
```

### CODE REVIEW

**Comment by edgarlanting** [comment]
Hi @ajanikow can you review?

File: examples/production-cluster-with-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+apiVersion: "database.arangodb.com/v1alpha"
```
Comment (ajanikow) [inline examples/production-cluster-with-metrics.yaml:1]: Can you switch to `database.arangodb.com/v1`? Even, you can do it in all example files

File: examples/production-cluster-with-metrics.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+metadata:
+  name: "production-cluster"
+spec:
+  metrics:
+    mode: sidecar
+    enabled: true
+    image: arangodb/arangodb-exporter:0.1.7
+    tls: false
+  annotations:                                                                                                                                                  
+    prometheus.io/scrape: 'true'
```
Comment (ajanikow) [inline examples/production-cluster-with-metrics.yaml:12]: Please - use ' or " quotes everywhere

File: examples/production-cluster-with-metrics.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    mode: sidecar
+    enabled: true
+    image: arangodb/arangodb-exporter:0.1.7
+    tls: false
+  annotations:                                                                                                                                                  
+    prometheus.io/scrape: 'true'                                                                                                                               
+    prometheus.io/port: '9101'                                                                                                                                 
+    prometheus.io/scrape_interval: '5s'
+  mode: Cluster
+  image: arangodb/arangodb:3.6.1
```
Comment (ajanikow) [inline examples/production-cluster-with-metrics.yaml:16]: Switch to more recent ArangoDB version (you can also do it in all files)

**Review by ajanikow — CHANGES_REQUESTED** [review]

File: examples/production-cluster-with-metrics.yaml (hunk 1 above)
Comment (edgarlanting) [reply examples/production-cluster-with-metrics.yaml:1]: Yes sure! Will take care of that!

File: examples/production-cluster-with-metrics.yaml (hunk 2 above)
Comment (edgarlanting) [reply examples/production-cluster-with-metrics.yaml:12]: Yes will do

File: examples/production-cluster-with-metrics.yaml (hunk 3 above)
Comment (edgarlanting) [reply examples/production-cluster-with-metrics.yaml:16]: Yes will do!

**Comment by edgarlanting** [comment]
@ajanikow I'll close this PR and accommodate all changes in a neat new PR

**Pull request closed without merging**